sklearn/linear_model/_omp.py

```diff
@@ -1071,18 +1075,34 @@ def fit(self, X, y):
         y : array-like of shape (n_samples,)
             Target values. Will be cast to X's dtype if necessary.
 
+        **fit_params : dict
+            Parameters to pass to the underlying splitter.
+
+            .. versionadded:: 1.4
+                Only available if `enable_metadata_routing=True`. See
```

Comment:
Could you add a reference to `set_config` and `config_context` as well to know that the parameter is set in those.